internal/declarative/v2/reconciler.go

```diff
@@ -609,9 +609,24 @@ func (r *Reconciler) configClient(ctx context.Context, obj Object) (Client, erro
 	if err != nil {
 		return nil, err
 	}
+
+	err = testClient(ctx, clnt)
```

Comment:
it's not a good approach test client connection inside the configuration, the purpose of this `configClient` is just for config client, but not take care if the connection is fine or not. To handle the connection error is the response in the follow-up process, but not here.

Reply:
ok the function call has been moved. I decided to keep it in the getTargetClient function as the error value from that funciton creates a ClientIntialization error event. which is fitting because if testing the client fails, it should count as initialization failure